### PR TITLE
Fixed web export of SVG and PNG

### LINKED ISSUE #682: Web build exports an unknown file
Whenever you export a file in the web build, it exports a file called "Unknown" for SVG and "png" for PNG without a file extension. Simply changing these files to add a .svg or .png seems to fix it but it still would be nice to be fixed.

### PR DESCRIPTION
**Pull request by Qainguin**
This fixes #682 by adding MIME types to the JavaScriptBridge and setting a default name if there is no file name in GlobalSettings.

### CODE REVIEW

**Comment by MewPurPur** [comment]
Thank you <3

How do you want to be mentioned in Authors.MD? Just Qainguin or also your real name?

File: src/ui_parts/export_dialog.tscn [hunk 1]
```diff
@@ -103,7 +102,6 @@ allow_higher = false
 unique_name_in_owner = true
 layout_mode = 2
 theme_override_colors/font_color = Color(0.666667, 0.666667, 0.666667, 1)
-theme_override_fonts/font = ExtResource("4_4t2iq")
```
Comment (MewPurPur) [inline src/ui_parts/export_dialog.tscn:106]: This change seems to be unintentional?

Comment (Kiisu-Master) [reply src/ui_parts/export_dialog.tscn:106]: The Godot uid bugs strike again..

**Comment by Qainguin** [comment]
> What's going on with the font import changing

Whenever I edit this repository with my Mac it just gets rid of the fallbacks.

File: src/ui_parts/export_dialog.tscn (hunk 1 above)
Comment (Qainguin) [reply src/ui_parts/export_dialog.tscn:106]: Yeah, I did not change this, Godot did it by itself.

**Comment by Kiisu-Master** [comment]
> Whenever I edit this repository with my Mac it just gets rid of the fallbacks.

What Godot version do you use? Also in git you can specify the files you want to commit changes of.

**Comment by Qainguin** [comment]
> > Whenever I edit this repository with my Mac it just gets rid of the fallbacks.
> 
> What Godot version do you use? Also in git you can specify the files you want to commit changes of.

I am using 4.2.2 stable. Also, thanks for letting me know that.

File: src/ui_parts/export_dialog.tscn (hunk 1 above)
Comment (Qainguin) [reply src/ui_parts/export_dialog.tscn]: Alright, I reverted the changes to unrelated files.

**Comment by Qainguin** [comment]
Ok, I have reverted all changes to unrelated files, it should now be ready for merge.

File: src/HandlerGUI.gd [hunk 2]
```diff
@@ -315,13 +315,23 @@ func _import(svg_text: String, file_name: String) -> void:
...
+				"export", 
+				"image/svg+xml")
 
 func web_save_png(img: Image) -> void:
-	JavaScriptBridge.download_buffer(img.save_png_to_buffer(),
-			Utils.get_file_name(GlobalSettings.save_data.current_file_path) + ".png")
-
+	if GlobalSettings.save_data.current_file_path:
+		JavaScriptBridge.download_buffer(img.save_png_to_buffer(),
+				Utils.get_file_name(GlobalSettings.save_data.current_file_path) + ".png")
```
Comment (Kiisu-Master) [inline src/HandlerGUI.gd:330]: MIME missing?

**Comment by Kiisu-Master** [comment]
Maybe to reduce code duplication:
```gdscript

func web_save_svg() -> void:
	var file_name := Utils.get_file_name(GlobalSettings.save_data.current_file_path)
	if file_name.is_empty():
		file_name = "export"
	JavaScriptBridge.download_buffer(
		SVG.text.to_utf8_buffer(),
		file_name + ".svg",
		"image/svg+xml"
	)


func web_save_png(img: Image) -> void:
	var file_name := Utils.get_file_name(GlobalSettings.save_data.current_file_path)
	if file_name.is_empty():
		file_name = "export"
	JavaScriptBridge.download_buffer(
		img.save_png_to_buffer(),
		file_name + ".png",
		"image/png"
	)

```

**Comment by Qainguin** [comment]
I added those changes as well as getting rid of adding the file extensions. (MIME already adds the file extension)

**Comment by Qainguin** [comment]
Screenshot 2024-04-26 at 11 21 54 AM
Is the d7 a separate issue?

**Comment by MewPurPur** [comment]
Is this ready to be merged? If you want, you can include the font change we discussed and close #685

**Comment by Qainguin** [comment]
It's ready to be merged but I'm gonna add the font change, give me like 15 minutes.

**Comment by Qainguin** [comment]
It is now ready to be merged, with the fix to #685.

**Pull request merged**